builder/source/simple.go

```diff
@@ -48,9 +48,15 @@ func NewSimple(uri, validator string, legacy bool) (*SimpleSource, error) {
 	if err != nil {
 		return nil, err
 	}
+	fileName := filepath.Base(uriObj.Path)
+	if uriObj.Fragment != "" {
+		fileName = uriObj.Fragment
```

Comment:
Haven't taken a good look yet but maybe you need to`filepath.Base(uriObj.Fragment)` ?

Reply:
Tested it, it is not necessary.